R/refineCenters.R

```diff
@@ -0,0 +1,143 @@
+#' @title Refine Peak Centers
+#'
+#' @description
+#' This function refines the center values of a peak by weighting the y values
+#' in the neightbourhood that belong most likely to the same peak.
+#'
+#' @param x `numeric`, i.e. m/z values.
+#' @param y `numeric`, i.e. intensity values.
+#' @param p `numeric`, peak centers.
+#' @param k `integer(1)`, number of values left and right of the peak that
+#'  should be considered in the weighted mean calculation.
+#' @param threshold `double`, ratio of the maximal peak intensity. Just values
+#'  above are used for the weighted mean calclulation.
+#' @param descending `logical`, if `TRUE` just values between the nearest
```

Comment:
So I guess you are no longer *walking* down the peak as I did, but use signal between the *peak* and the closest *valley*?

Reply:
That's right.